packages/back-end/src/routers/environment/environment.router.ts

```diff
@@ -29,8 +30,40 @@ router.put(
   environmentController.putEnvironments
 );
 
+router.put(
+  "/order",
+  validateRequestMiddleware({
+    body: z
+      .object({
+        environments: z.array(z.string()),
+      })
+      .strict(),
+  }),
+  environmentController.putEnvironmentOrder
+);
+
 router.post(
   "/",
+  validateRequestMiddleware({
+    body: z
+      .object({
+        environment: z
+          .object({
+            id: z.string(),
+            description: z.string(),
+            toggleOnList: z.boolean().optional(),
+            defaultState: z.any().optional(),
+            projects: z.array(z.string()).optional(),
+          })
+          .strict(),
```

Comment:
Should this be factored out to make it more visible for potential future re-use?